docs/building_pipelines.rst

```diff
@@ -293,3 +294,423 @@ Warnings raised will also be logged to the error log but will not cause data to
 
 See :ref:`AccessingLogs`: for details have how to access the system logs.
 
+Debugging & Tracing Pipelines
+-----------------------------
+
+Fledge has a feature that allows the debugging of pipelines in south an north services. It provides a mechanism to view the data as it flows through the pipeline.
+
+The debugger is designed to show the data as it traverses the pipeline within the service. Users may;
+
+   - Show the data at each stage in the pipeline.
+
+   - Configure the number of readings to store at each stage in the pipeline.
+
+   - Pause the ingest into the pipeline.
+
+   - Suspend the output of the pipeline.
+
+   - Ingest a number of readings into the paused pipeline.
+
+   - Replay the currently buffered readings for a pipeline.
+
+   - Resume flow into and out of the pipeline.
+
+
+The *replay* operation is useful when manipulating the configuration of the pipeline components. The user may change a filter configuration, replay the saved readings to see the impact of the configuration change. This may be repeated multiple times until the user is then satisfied with the result of the filter configuration.
```

Comment:
- filter configuration, replay the saved readings -> filter configuration and replay the saved readings
- until the user is then satisfied -> until the user is satisfied